The backend does not require a mesonbuild module for every flavor.
the `meson` command from the regular RPM package is sufficient.

Index: meson_python-0.16.0/pyproject.toml
===================================================================
--- meson_python-0.16.0.orig/pyproject.toml
+++ meson_python-0.16.0/pyproject.toml
@@ -6,8 +6,6 @@
 build-backend = 'mesonpy'
 backend-path = ['.']
 requires = [
-  'meson >= 0.63.3; python_version < "3.12"',
-  'meson >= 1.2.3; python_version >= "3.12"',
   'packaging >= 19.0',
   'pyproject-metadata >= 0.7.1',
   'tomli >= 1.0.0; python_version < "3.11"',
@@ -33,8 +31,6 @@ classifiers = [
 ]
 
 dependencies = [
-  'meson >= 0.63.3; python_version < "3.12"',
-  'meson >= 1.2.3; python_version >= "3.12"',
   'packaging >= 19.0',
   'pyproject-metadata >= 0.7.1',
   'tomli >= 1.0.0; python_version < "3.11"',
